action/rlp_tx.go

```diff
@@ -82,35 +87,150 @@ func reconstructSignedRlpTxFromSig(tx rlpTransaction, chainID uint32, sig []byte
 }
 
 // DecodeRawTx decodes raw data string into eth tx
-func DecodeRawTx(rawData string, chainID uint32) (tx *types.Transaction, sig []byte, pubkey crypto.PublicKey, err error) {
+func DecodeRawTx(rawData string, chainID uint32) (*types.Transaction, bool, error) {
 	//remove Hex prefix and decode string to byte
-	rawData = strings.Replace(rawData, "0x", "", -1)
-	rawData = strings.Replace(rawData, "0X", "", -1)
-	var dataInString []byte
-	dataInString, err = hex.DecodeString(rawData)
+	dataInString, err := hex.DecodeString(util.Remove0xPrefix(rawData))
 	if err != nil {
-		return
+		return nil, true, err
 	}
 
 	// decode raw data into rlp tx
-	tx = &types.Transaction{}
-	err = rlp.DecodeBytes(dataInString, tx)
+	unwrapper := &transactionUnwrapper{}
+	if err = rlp.DecodeBytes(dataInString, unwrapper); err != nil {
+		return nil, true, err
+	}
+	return unwrapper.tx, unwrapper.isEthEncoding, nil
+}
+
+// EncodeRawTx encodes action into the data string of eth tx
+func EncodeRawTx(act Action, pvk crypto.PrivateKey, chainID uint32) (string, error) {
+	rlpAct, err := actionToRLP(act)
+	if err != nil {
+		return "", err
+	}
+	rawTx, err := rlpToEthTx(rlpAct)
 	if err != nil {
-		return
+		return "", err
+	}
+	ecdsaPvk, ok := pvk.EcdsaPrivateKey().(*ecdsa.PrivateKey)
+	if !ok {
+		return "", errors.New("private key is invalid")
 	}
 
-	// extract signature and recover pubkey
+	signer := types.NewEIP155Signer(big.NewInt(int64(chainID)))
+	signedTx, err := types.SignTx(rawTx, signer, ecdsaPvk)
+	if err != nil {
+		return "", err
+	}
+	encodedTx, err := rlp.EncodeToBytes(signedTx)
+	if err != nil {
+		return "", err
+	}
+	return hex.EncodeToString(encodedTx[:]), nil
+}
+
+// ExtractSignatureAndPubkey calculates signature and pubkey from tx
+func ExtractSignatureAndPubkey(tx *types.Transaction, pbAct *iotextypes.ActionCore, chainID uint32, isEthEncoding bool) ([]byte, crypto.PublicKey, error) {
```

Comment:
wrong place to handle the logic. 
1. This is supposed to be at the api level
2. signature has nothing to do with pbAct